### PR TITLE
Move ApiObjects to JsonApi namespace/folder

### PR DESCRIPTION
**Pull request by groyoh**
Follow up to
https://github.com/rails-api/active_model_serializers/pull/1504/files#r52796473.
Moved the Relationship and ResourceIdentifier to the JsonApi
namespace/folder and slightly refactored the Relationship code.

### CODE REVIEW

File: lib/active_model/serializer/adapter/json_api/relationship.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        class Relationship
```
Comment (groyoh) [inline lib/active_model/serializer/adapter/json_api/relationship.rb]: @bf4 following your [comment](https://github.com/rails-api/active_model_serializers/pull/1504/files#r52563899) I was not sure if they should still be namespace by `ApiObjects` or not.

**Comment by groyoh** [comment]
The tests files were not changed/moved because they would clash with some other test name (e.g. `test/adapter/json_api/relationship_test.rb`) so I kept them under `test/adapter/json_api/api_objects` folder.

File: lib/active_model/serializer/adapter/json_api/relationship.rb (hunk 1 above)
Comment (remear) [inline lib/active_model/serializer/adapter/json_api/relationship.rb]: namespacing it under `ApiObjects` seems unnecessary to me.

**Comment by remear** [comment]
What if the tests from `test/adapter/json_api/api_objects` were merged into the files of the same name in the parent directory?

**Comment by groyoh** [comment]
And have two different group of tests in the same test file? I don't think that's appropriate. To be clear here `test/adapter/json_api/relationship_test.rb` tests the jsonapi adapter behavior when using relationship and `test/adapter/json_api/api_objects/relationship_test.rb` tests the relationship object itself.
Maybe there is a better solution here.
________________________________

**Comment by remear** [comment]
Do you have a preference on how you'd like it to be structured?

**Comment by groyoh** [comment]
@remear not really. Any suggestion would be good to me. I would just not like to merge the two different concerns into the same file.

**Comment by groyoh** [comment]
@bf4 rebased on master and moved it to `ActiveModelSerializers` namespace.

File: lib/active_model/serializer/adapter/json_api/relationship.rb (hunk 1 above)
Comment (bf4) [inline lib/active_model/serializer/adapter/json_api/relationship.rb]: Agreed. Under JsonApi is fine, see my errors pr

File: test/adapter/json_api/api_objects/resource_identifier_test.rb [hunk 2]
```diff
@@ -1,87 +1,84 @@
 require 'test_helper'
```
Comment (bf4) [inline test/adapter/json_api/api_objects/resource_identifier_test.rb]: Let's move these downs down a level and delete the api_objects directory

**Comment by groyoh** [comment]
Closing in favor of #1543

**Pull request closed without merging**